Add GKE Autopilot support for Datadog CSI Driver (#32253)
DataDog/documentation: content/en/containers/csi_driver/_index.md

{{< /tabs >}}

> ### GKE Autopilot Support

Reviewer: ```suggestion
### GKE Autopilot support
```

> Starting with **Helm chart version 3.138.0**, the Datadog CSI Driver can be installed on **Google Kubernetes Engine (GKE) Autopilot** clusters.

Reviewer 2: Would be better to clarify which chart we are referring to. 

The minimum datadog agent chart version is 3.138.0.
The minimum datadog csi driver chart version is 0.4.2.

> Starting with **Helm chart version 3.138.0**, the Datadog CSI Driver can now be installed on **Google Kubernetes Engine (GKE) Autopilot** clusters.

Reviewer: ```suggestion
Starting with **Helm chart version 3.138.0**, the Datadog CSI Driver can be installed on **Google Kubernetes Engine (GKE) Autopilot** clusters.
```

> **Note:**  

> > GKE Autopilot support is **only available when installing via the Helm chart**.  

Reviewer: ```suggestion
> GKE Autopilot support is **only available when installing using the Helm chart**.  
```

> > For non-Helm installations, render and apply the CSI Driver manifests manually, then create the following `AllowlistSynchronizer` resource to enable installation on GKE Autopilot:

Reviewer 2: ```suggestion
> If CSI driver is not installed using the helm chart, create the following `AllowlistSynchronizer` resource to activate GKE Autopilot support:
```
